### PR TITLE
Mentor: accept reject mentorship requests

### LINKED ISSUE #61: Mentor: accept/reject mentorship request
### Feature Description

- add endpoints acceptMentorship, rejectMentorship
- add handling the accept button and reject dropdown to send the request to the right endpoint
- add toast at the end of the operation

Accept mentorship endpoint: 
- get the request, set its status to accepted
- create a mentorship relationship duplicating data
- send mentorship accepted notification email

Reject mentorship endpoint:
- get the request, set its status to rejected, set the reason
- send rejected mentorship notification email, with the reason

### Screenshot

mentor-accept-reject-mentorship

### PR DESCRIPTION
**Pull request by lorepirri**
### What Github issue does this PR relate to?

Close #61 

Changes proposed in this pull request:

- accept / reject mentorship

### What should the reviewer know?

- sits on top of `feat/58-mentor_contact_mentee`
- email notifications depend on PR #53 and PR #51 

mentor-accept-reject-mentorship

---

image

### Quality Assurance(QA) Checklist

- [x] The app builds and runs properly

### CODE REVIEW

**Comment by lorepirri** [comment]
I did `rebase feat/58-mentor_contact_mentee` and `push --force`
please `pull --rebase`

**Review by tsg — APPROVED** [review]
Nice, left just a nit-pick, feel free to ignore.

File: pages/api/rejectMentorship.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  // just make sure that the request has a valid status
+  let requestStatus = RequestStatusEnum.Rejected;
+  switch (req.body.requestStatus) {
+    case RequestStatusEnum.Rejected:
+    case RequestStatusEnum.RejectedNoGoodFit:
+    case RequestStatusEnum.RejectedBusy:
+      requestStatus = req.body.requestStatus;
+      break;
+    default:
+      requestStatus = RequestStatusEnum.Rejected;
```
[the remark this answers is not in the document]
Comment (lorepirri) [reply pages/api/rejectMentorship.js]: this is very useful, thanks. I actually added the validator.

I did not use `Symbol` for the values in the Enums as then it makes delicate to compare string status to enum status.

See: https://www.sohamkamani.com/javascript/enums/

**Comment by lorepirri** [comment]
in 086faf7 I've added retrieving only pending requests. Before was not so, because originally we wanted to show all the requests even the past requests and sort them by status. At the moment this is not done for the MVP, so it will done during that follow up.

**Pull request merged**